<launch>
  <arg name="load_manager" default="true"/>
  <arg name="manager_name" default="camera_manager"/>
  <arg name="manager" value="/$(arg manager_name)"/>
  <arg name="manager_threads" default="4"/>

  <arg name="camera_name" default="web_cam"/>

  <!-- Nodelet Manager -->
  <node if="$(arg load_manager)" pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
    <param name="num_worker_threads" value="$(arg manager_threads)" />
  </node>

  <!-- USB Cam -->
  <node pkg="usb_cam" type="usb_cam_node" name="$(arg camera_name)" args="load camera_aravis/CameraAravisNodelet $(arg manager)" output="screen">
    <rosparam command="load" file="$(find camera_tracking)/config/$(arg camera_name).yaml" />
    <param name="frame_id" value="$(arg camera_name)" />
    <param name="camera_info_url" value="package://camera_tracking/config/$(arg camera_name)_calibration.yaml" />
  </node>

  <!-- tf Remapping -->
  <node pkg="camera_tracking" type="tf_rebroadcaster" ns="$(arg camera_name)" name="tf_rebroadcaster" output="screen">
    <param name="prefix" value="$(arg camera_name)" />
  </node>
  <node pkg="camera_tracking" type="camera_origin" ns="$(arg camera_name)" name="camera_origin" output="screen">
    <param name="camera_name" value="$(arg camera_name)" />
    <param name="origin_id" value="tag_0" />
  </node>

  <!-- De-Bayer / rectify -->
  <node pkg="image_proc" type="image_proc" ns="$(arg camera_name)" name="image_proc" respawn="false" output="screen"/>

  <!-- Apriltag detection -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" ns="$(arg camera_name)" name="apriltag" output="screen">
    <param name="camera_frame" type="str" value="$(arg camera_name)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />

    <rosparam command="load" file="$(find camera_tracking)/config/settings.yaml" />
    <rosparam command="load" file="$(find camera_tracking)/config/tags.yaml" />

    <remap from="/tf" to="tf" />
  </node>
</launch> 